src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -1885,36 +1885,157 @@ public override void VisitMarkupElement(MarkupElementSyntax node)
             {
                 _document.Diagnostics.Add(
                     ComponentDiagnosticFactory.Create_UnsupportedComponentImportContent(BuildSourceSpanFromNode(node)));
+
+                base.VisitMarkupElement(node);
             }
 
             public override void VisitMarkupCommentBlock(MarkupCommentBlockSyntax node)
             {
                 _document.Diagnostics.Add(
                     ComponentDiagnosticFactory.Create_UnsupportedComponentImportContent(BuildSourceSpanFromNode(node)));
+
+                base.VisitMarkupCommentBlock(node);
             }
 
             public override void VisitMarkupTagHelperElement(MarkupTagHelperElementSyntax node)
             {
                 _document.Diagnostics.Add(
                     ComponentDiagnosticFactory.Create_UnsupportedComponentImportContent(BuildSourceSpanFromNode(node)));
+
+                base.VisitMarkupTagHelperElement(node);
+            }
+
+            public override void VisitCSharpExplicitExpression(CSharpExplicitExpressionSyntax node)
+            {
+                if (node.FirstAncestorOrSelf<SyntaxNode>(n => n is MarkupStartTagSyntax || n is MarkupEndTagSyntax) != null)
```

Comment:
Why is `base.VisitCSharpExplicitExpresion` not sufficient for all of this?

Reply:
Because base isn't `ComponentFileKindVisitor`.